### PR TITLE
fix(phemex) - funding fees

### LINKED ISSUE #22434: Phemex: fetchFundingHistory -> amount: null
### Operating System

Debian 12

### Programming Languages

Python

### CCXT Version

4.3.18

### Description

Moin,

Method fetchFundingHistory gives "amount: null" on Phemex.
Seems wrong for me.

### Code

```
print(exchange.fetchFundingHistory('BTC/USDT:USDT'))
```
Example Result:
```
    "info": {
      "symbol": "BTCUSDT",
      "currency": "USDT",
      "execQtyRq": "0.001",
      "side": "Buy",
      "execPriceRp": "64905.93672508",
      "execValueRv": "64.90593673",
      "fundingRateRr": "0.00003809",
      "feeRateRr": "0.00003809",
      "execFeeRv": "0.00247227",
      "createTime": "1713744000000"
    },
    "symbol": "BTCUSDT",
    "code": "USDT",
    "timestamp": 1713744000000,
    "datetime": "2024-04-22T00:00:00.000Z",
    "id": null,
    "amount": null
  }
```
Amount should not be null I think,

### PR DESCRIPTION
**Pull request by ttodua**
fix #22434

### CODE REVIEW

File: ts/src/phemex.ts [hunk 1]
```diff
@@ -3837,6 +3837,7 @@ export default class phemex extends Exchange {
             request['limit'] = limit;
         }
         let response = undefined;
+        this.verbose = true;
```
Comment (carlosmiei) [inline ts/src/phemex.ts:3840]: @ttodua leftover

**Comment by carlosmiei** [comment]
@ttodua something is off here 
```
n phemex fetchFundingHistory "BTC/USD:USD"  --sandbox  
2024-05-09T21:18:08.520Z
Node.js: v18.18.0
CCXT v4.3.18
phemex.fetchFundingHistory (BTC/USD:USD)
2024-05-09T21:18:11.072Z iteration 0 passed in 856 ms

 symbol | code |     timestamp |                 datetime | amount
------------------------------------------------------------------
uBTCUSD |  USD | 1664409600000 | 2022-09-29T00:00:00.000Z |  11452
uBTCUSD |  USD | 1664380800000 | 2022-09-28T16:00:00.000Z |  11695
uBTCUSD |  USD | 1664352000000 | 2022-09-28T08:00:00.000Z |  11270
uBTCUSD |  USD | 1664323200000 | 2022-09-28T00:00:00.000Z |     39
uBTCUSD |  USD | 1664294400000 | 2022-09-27T16:00:00.000Z |     40
uBTCUSD |  USD | 1664265600000 | 2022-09-27T08:00:00.000Z |     41
uBTCUSD |  USD | 1664236800000 | 2022-09-27T00:00:00.000Z |
```

File: ts/src/phemex.ts (hunk 1 above)
Comment (ttodua) [reply ts/src/phemex.ts]: ohh give me a sec :(

**Comment by ttodua** [comment]
I'll try to open and retain a position for a day or so, as I didnt have enough timeframe to observe in realtime.

**Comment by ttodua** [comment]
pushed a fix for USD contracts, now waiting for BTC (Inverse) funding event to confirm that fix also works for inverse markets

**Comment by carlosmiei** [comment]
@ttodua any luck?

**Comment by ttodua** [comment]
> @ttodua any luck?

I've got unexpected results, so i wrote to phemex support and waiting their response again, will probably have final answer soon

**Comment by ttodua** [comment]
should be ready. tested against `BTC/USD:USD`, `BTC/USDT:USDT`, `BTC/USD:BTC`

**Comment by ttodua** [comment]
@carlosmiei ^

**Comment by carlosmiei** [comment]
> @carlosmiei ^

@ttodua thanks will merge it shortly

**Comment by carlosmiei** [comment]
DEMO

```
p phemex fetchFundingHistory "BTC/USDT:USDT" --sandbox
Python v3.11.7
CCXT v4.3.18
phemex.fetchFundingHistory(BTC/USDT:USDT)
[{'amount': '0.0642651',
  'code': 'USDT',
  'datetime': '2024-04-25T00:00:00.112Z',
  'id': None,
  'info': {'createTime': '1714003200112',
           'currency': 'USDT',
           'execFeeRv': '0.0642651',
           'execPriceRp': '64265.1',
           'execQtyRq': '0.01',
           'execValueRv': '642.651',
           'feeRateRr': '0.0001',
           'fundingRateRr': '0.0001',
           'side': 'Buy',
           'symbol': 'BTCUSDT'},
  'symbol': 'BTCUSDT',
  'timestamp': 1714003200112},
```
```
p phemex fetchFundingHistory "BTC/USD:USD" --sandbox 
Python v3.11.7
CCXT v4.3.18
phemex.fetchFundingHistory(BTC/USD:USD)
[{'amount': '1.1452',
  'code': 'USD',
  'datetime': '2022-09-29T00:00:00.000Z',
  'id': None,
  'info': {'createTime': '1664409600000',
           'currency': 'USD',
           'execFeeEv': '11452',
           'execPriceEp': '194132173',
           'execQty': '2',
           'execValueEv': '388264',
           'feeRateEr': '2949500',
           'fundingRateEr': '2949500',
           'side': 'Buy',
           'symbol': 'uBTCUSD'},
  'symbol': 'uBTCUSD',
  'timestamp': 1664409600000},
 {'amount': '1.1695',
  'code': 'USD',
  'datetime': '2022-09-28T16:00:00.000Z',
  'id': None,
```

**Pull request merged**